std_execution.bs

```diff
@@ -3582,29 +3585,71 @@ are all well-formed.
 #### `execution::when_all` <b>[execution.senders.adaptors.when_all]</b> #### {#spec-execution.senders.adaptor.when_all}
 
 1. `execution::when_all` is used to join multiple sender chains and create a sender whose execution is dependent on all of the input senders that only send a single set of values. `execution::when_all_with_variant`
-    is used to join multiple sender chains and create a sender whose execution is dependent on all of the input senders, which may have one or more sets of sent values.
+    is used to join multiple sender chains and create a sender whose execution is dependent on all of the input senders, each of which may have one or more sets of sent values.
+
+2. The name `execution::when_all` denotes a customization point object. For some subexpressions <code>s<i><sub>i</sub></i>...</code>, let <code>S<i><sub>i</sub></i>...</code> be <code>decltype((s<i><sub>i</sub></i>))...</code>. The expression <code>execution::when_all(s<i><sub>i</sub></i>...)</code> is ill-formed if any of the following is true:
+
+    * If the number of subexpressions <code>s<i><sub>i</sub></i>...</code> is 0, or
+    * If any type <code>S<i><sub>i</sub></i></code> does not satisfy `execution::typed_sender`, or
+    * If for any type <code>S<i><sub>i</sub></i></code>, the type <code>sender_traits&lt;remove_cvref_t&lt;S<i><sub>i</sub></i>>>::value_types&lt;tuple, <i>zero-or-one</i>></code> is ill-formed, where <code><i>zero-or-one</i></code> is a template alias equivalent to the following:
+        <pre highlight="c++">
+        template &lt;class... Ts>
+            requires (sizeof...(Ts) <= 1)
+          using <i>zero-or-one</i> = void;
+        </pre>
+
+    Otherwise, the expression <code>execution::when_all(s<i><sub>i</sub></i>...)</code> is expression-equivalent to:
+
+    1. <code>tag_invoke(execution::when_all, s<i><sub>i</sub></i>...)</code>, if that expression is valid and its type satisfies `execution::sender`. If the function selected by `tag_invoke` does not return a sender that sends a concatenation of values sent by <code>s<i><sub>i</sub></i>...</code> when they all complete with `set_value`, the program is ill-formed with no diagnostic required.
+
+    2. Otherwise, constructs a sender `w` of type `W`. When `w` is connected with some receiver `out_r` of type `OutR`, it returns an operation state `op_state` specified as below:
```

Comment:
Drop the part about op_state here, since it's spelled out below?

Reply:
I added it here because I mention `op_state` in the specification of the inner receivers' semantics, before I introduce `op_state` below.